lib/rules/identical-links-same-purpose-matches.js

```diff
@@ -0,0 +1,79 @@
+const { aria, text, dom } = axe.commons;
+const { escapeSelector } = axe.utils;
+
+/**
+ * Note:
+ * Firefox does not respect map > area as visible,
+ * even when used by a img[usemap], hence the usage of `includeHidden` flag,
+ * in the accessible name computation
+ */
+const hasAccName = !!text.accessibleTextVirtual(virtualNode, {
+	includeHidden: true
+});
+if (!hasAccName) {
+	return false;
+}
+
+const role = aria.getRole(node);
+if (role && role !== 'link') {
+	return false;
+}
+
+const nodeName = node.nodeName.toUpperCase();
+if (['A', 'AREA'].includes(nodeName) && !node.hasAttribute('href')) {
+	return false;
+}
+
+if (nodeName === 'AREA') {
+	const rootEl = dom.getRootNode(node);
+	/**
+	 * Ensure map is used by `img` witin the document tree
+	 */
+	const mapEl = getMapForAreaElement(node, rootEl);
+	if (!mapEl) {
+		return false;
+	}
+
+	const mapElName = mapEl.getAttribute('name');
+	if (!mapElName) {
+		return false;
+	}
+
+	return isMapReferredByAnyImgNode(rootEl, mapElName);
+}
+
+return true;
+
+/**
+ * Check if a given name is referred by <img> node within the `usemap` attribute
+ * @param {HTMLElement} rootEl root node (document or shadowRoot)
+ * @param {String} mapElName element name
+ * @reurns {Boolean}
+ */
+function isMapReferredByAnyImgNode(rootEl, mapElName) {
+	const mapName = escapeSelector(mapElName);
+	const refs = rootEl.querySelectorAll(`[usemap="#${mapName}"]`);
```

Comment:
You should probably just limit this to `img[usemap="..."]`. More accurate and speeds up the selector.